### PR TITLE
PSR-462 Added new address lookup flow

### PR DESCRIPTION
**Pull request by westwater**

### CODE REVIEW

File: app/config/Constants.scala [hunk 1]
```diff
@@ -62,11 +62,16 @@ object Constants {
...
+  val maxInputLength = 35
   val maxTextAreaLength = 160
   val maxTitleNumberLength = 35
 
   val textAreaRegex = """^[a-zA-Z0-9\-'" \t\r\n,.@/]+$"""
   val titleNumberRegex = """^[a-zA-Z]{2,3}[0-9]+$"""
 
+  val postcodeCharsRegex = """^[a-zA-Z0-9 ]+$"""
+  val postcodeFormatRegex =
+    """^(GIR 0AA)|((([a-zA-Z-[QVX]][0-9][0-9]?)|(([a-zA-Z-[QVX]][a-zA-Z-[IJZ]][0-9][0-9]?)|(([a-zA-Z-[QVX]][0-9][A-HJKSTUW])|([a-zA-Z-[QVX]][a-zA-Z-[IJZ]][0-9][ABEHMNPRVWXY]))))\s?[0-9][a-zA-Z-[CIKMOV]]{2})$"""
```
Comment (kberec) [inline app/config/Constants.scala:74]: Should this match the one in ETMP?

File: conf/messages.en [hunk 2]
```diff
@@ -1555,4 +1555,51 @@ unallocatedEmployerCYA.section.hide = amount of unallocated employer contributio
...
+landPropertyAddressResults.description.link = enter the address manually.
+landPropertyAddressResults.error.required = Select the address of the land or property
+
+landPropertyAddressManual.title = What is the address of the land or property?
+landPropertyAddressManual.heading = What is the address of the land or property?
+landPropertyAddressManual.field1.label = Address line 1
+landPropertyAddressManual.field1.error.required = Enter the first line of the address
+landPropertyAddressManual.field1.error.invalid = First line of address must only include letters a to z, numbers 0 to 9, apostrophes, ‘atʼ signs (@), commas, forward slashes, full stops, hyphens, speech marks and spaces
+landPropertyAddressManual.field1.error.max = First line of the address must be 35 characters or fewer
+landPropertyAddressManual.field2.label = Address line 2 (optional)
```
Comment (kberec) [inline conf/messages.en:1586]: Mandatory in ETMP

File: .g8/moneyPage/app/controllers/nonsipp/$if(directory.empty)$.$else$$directory$$endif$/$className$Controller.scala [hunk 3]
```diff
@@ -16,6 +16,7 @@
 
 $! Generic imports !$
 $if(directory.empty)$
+
```
Comment (kberec) [inline .g8/moneyPage/app/controllers/nonsipp/$if(directory.empty)$.$else$$directory$$endif$/$className$Controller.scala:19]: Are these .g8 changes related to the addressLookup?

File: app/models/Address.scala [hunk 4]
```diff
@@ -16,55 +16,135 @@
...
+    val (addressLine1, addressLine2, addressLine3, town, postCode, country) = tup
+    Address(
+      "manual",
+      addressLine1,
+      addressLine2,
+      addressLine3,
+      town,
+      postCode,
+      country,
+      "N/A", // TODO get country code from country
```
Comment (kberec) [inline app/models/Address.scala:146]: Is this for later on? If so do we have a ticket for it?

File: app/config/Constants.scala (hunk 1 above)
Comment (westwater) [reply app/config/Constants.scala:74]: changed 👍

File: .g8/moneyPage/app/controllers/nonsipp/$if(directory.empty)$.$else$$directory$$endif$/$className$Controller.scala (hunk 3 above)
Comment (westwater) [reply .g8/moneyPage/app/controllers/nonsipp/$if(directory.empty)$.$else$$directory$$endif$/$className$Controller.scala:19]: yeah, to create some of the pages

File: app/models/Address.scala (hunk 4 above)
Comment (westwater) [reply app/models/Address.scala:146]: I remember Michael, Josh and I had a meeting around this - pretty sure Micheal created a ticket but just asked

File: conf/messages.en (hunk 2 above)
Comment (westwater) [reply conf/messages.en:1587]: Will change 👍

Comment (westwater) [reply conf/messages.en:1587]: leaving it as is as per the slack discussion https://hmrcdigital.slack.com/archives/C042149LQFN/p1699362656089599

**Review by kberec — APPROVED** [review]

**Pull request merged**